dotCMS/src/main/java/com/dotcms/vanityurl/filters/VanityUrlRequestWrapper.java

```diff
@@ -36,12 +35,23 @@ public VanityUrlRequestWrapper(final HttpServletRequest request, final VanityUrl
         super(request);
 
         final boolean vanityHasQueryString = UtilMethods.isSet(vanityUrlResult.getQueryString());
-        
-        this.newQueryString = vanityHasQueryString && UtilMethods.isSet(request.getQueryString())
-                        ? request.getQueryString() + "&" + vanityUrlResult.getQueryString()
-                        : vanityHasQueryString 
-                            ? vanityUrlResult.getQueryString()
-                            : request.getQueryString();
+
+        final StringBuilder params = new StringBuilder();
```

Comment:
It would be great to have a var query3 = RequestUtil.mergeQueryString(query1, query1)